### PR TITLE
Fix extra bracket in template literal syntax highlighting (#17327)

### PR DESCRIPTION
**Pull request by robobun**
## Summary

Fixes an issue where the JavaScript syntax highlighter was incorrectly adding an extra closing bracket (`}`) when highlighting template literal interpolations containing `JSON.stringify()` calls in colored terminal output.

## Problem

When running code like this with colors enabled:
```typescript
const result = {success:false};
const err = new Error(`error: ${JSON.stringify(result)}`);
throw err;
```

The error output would show an extra `}` character after `JSON.stringify(`, causing malformed syntax highlighting in the error display.

## Root Cause

The bug was in the template literal parsing code in `src/fmt.zig` where it would always write a closing brace after processing template literal interpolation, even when a closing brace was already found and consumed from the input.

**Before (buggy)**:
```zig
if (i < text.len and text[i] == '}') {
    i += 1;  // consume the brace
}
try writer.writeAll("}");  // always write a brace (BUG\!)
```

**After (fixed)**:
```zig
if (i < text.len and text[i] == '}') {
    try writer.writeAll("}");  // only write if found
    i += 1;  // then consume it
}
```

## Solution

- Only write the closing `}` character when it's actually found in the input
- This ensures the syntax highlighter doesn't add extra brackets to the output
- The fix is minimal and surgical, affecting only the specific case where the bug occurred

## Test plan

- [x] Manual reproduction confirmed the issue existed before the fix
- [x] Manual verification confirmed the extra bracket no longer appears after the fix
- [x] Created comprehensive regression test in `test/regression/issue/17327.test.ts`
- [x] Tested edge cases with nested JSON objects and arrays  
- [x] Verified fix works with both `FORCE_COLOR=1` and `NO_COLOR=1` modes
- [x] All tests pass locally

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

**Comment by robobun** [comment]
<div><sup>Updated 5:56 PM PT - Jul 18th, 2025</sup></div>

Your commit b8e762b0502935044f22a33f596b6fcbd564f110 is building: [`#20961`](https://buildkite.com/bun/bun/builds/20961)

File: test/regression/issue/17327.test.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import { expect, test } from "bun:test";
```
Comment (markovejnovic) [inline test/regression/issue/17327.test.ts:1]: Would potentially suggest making these tests shorter since, as it turns out, the issue is in the highlighting of interpolated strings specifically.

I've gotten minimum reproducibility with:

```js
import { highlightJavaScript as highlighter } from "bun:internal-for-testing";
import { expect, test } from "bun:test";
import stripAnsiColors from "strip-ansi";

test("highlighter-with-multiline-interpolated-string", () => {
  const testStr = '`foo: \${bar';
  const highlighted = highlighter(testStr);
  expect(stripAnsiColors(highlighted)).toBe(testStr);
});

test("highlighter-with-interpolated-string", () => {
  const testStr = '`foo: \${bar}';
  const highlighted = highlighter(testStr);
  expect(stripAnsiColors(highlighted)).toBe(testStr);
});
```
